src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -966,47 +966,76 @@ bool Literal::is_string_like() const noexcept {
 Literal Literal::add(Literal const &other, Node::NodeStorage &node_storage) const noexcept {
     return this->numeric_binop_impl([](auto const &num_ops) noexcept {
         return num_ops.add_fptr;
-    }, other, node_storage);
+    },
+                                    other, node_storage);
 }
 
 Literal Literal::operator+(Literal const &other) const noexcept {
     return this->add(other);
 }
 
+Literal &Literal::operator+=(const Literal &other) noexcept {
+    auto result = *this + other;
+    this->handle_ = result.handle_;
+    return *this;
+}
+
 Literal Literal::sub(Literal const &other, Node::NodeStorage &node_storage) const noexcept {
     return this->numeric_binop_impl([](auto const &num_ops) noexcept {
         return num_ops.sub_fptr;
-    }, other, node_storage);
+    },
+                                    other, node_storage);
```

Comment:
something strange happened to the formatting here (and some other places) 😅

Reply:
that seems to be what clang format wants it to look like

Reply:
Looks like a bug to me, I would understand this:
```c++
return this->numeric_binop_impl([](auto const &num_ops) noexcept {
				    return num_ops.sub_fptr;
    			        },
    			        other, node_storage);

```

but the thing it chose just looks broken.

Could you please try formatting it like I did and see if clang-format still wants to break it?
And regardless of what it does could you please file a bug report against clang-format? Just to make them aware of it. I mean it could be on purpose but I kinda doubt that

Reply:
seems like this is already reported: https://github.com/llvm/llvm-project/issues/58146
the suggested workaround formats it like this:
```cpp
    return this->numeric_binop_impl(
            [](auto const &num_ops) noexcept {
                return num_ops.add_fptr;
            },
            other, node_storage);
```
should i change them to the above or keep the current
```cpp
    // clang-format off
    return this->numeric_binop_impl([](auto const &num_ops) noexcept {
        return num_ops.add_fptr;
    }, other, node_storage);
    // clang-format on``` ?

Reply:
Please change it to the suggested workaround